### PR TITLE
feat: add DuckDB dialect

### LINKED ISSUE #873: Add DuckDB's integer devision operator `//`
Maybe related to #863 and #868

DuckDB introduced integer division operator `//` in the next release 0.8.0. (duckdb/duckdb#7082)
Is it worth supporting this here?

### PR DESCRIPTION
**Pull request by eitsupi**
Close #873

Add DuckDB dialect and support DuckDB's `//` operator.

### CODE REVIEW

**Comment by eitsupi** [comment]
Perhaps there is quite a lack of testing for dialects here.

I am not very experienced with this repository and DuckDB, so it would be great if someone could help me with this.

File: src/ast/operator.rs [hunk 1]
```diff
@@ -85,6 +85,7 @@ pub enum BinaryOperator {
     BitwiseOr,
     BitwiseAnd,
     BitwiseXor,
+    DuckIntegerDivide,
```
Comment (alamb) [inline src/ast/operator.rs:88]:
```suggestion
    /// Integer division operator `//` in DuckDB
    DuckIntegerDivide,
```

**Review by alamb — COMMENTED** [review]
Thank you for the contribution @eitsupi !

>  Perhaps there is quite a lack of testing for dialects here.

I think you need to add `DuckDbDialect` to the list in the following location and the new `DuckDbDialect` will be well covered by tests/sqlparser_common.rs

https://github.com/sqlparser-rs/sqlparser-rs/blob/4559d87a824a6a268a350d02e63c51b77e41bd34/src/test_utils.rs#L158-L173

File: src/tokenizer.rs [hunk 2]
```diff
@@ -768,6 +771,9 @@ impl<'a> Tokenizer<'a> {
                                 comment,
                             })))
                         }
+                        Some('/') if dialect_of!(self is DuckDbDialect) => {
```
Comment (alamb) [inline src/tokenizer.rs:774]: Can you also check here for `GenericDialect` as well?

**Comment by alamb** [comment]
This is looking pretty good @eitsupi  --  it has a conflict that needs to be resolved and I think the test needs a little updating but otherwise it is ready to go

**Review by alamb — APPROVED** [review]
Looks great to me -- thank you @eitsupi

**Pull request merged**